[i18n] Full i18n Support

dodoels opened the pull request:
# 🌍 Full i18n Support

![image](https://github.com/user-attachments/assets/c0a9cc87-0f33-4c28-a8fb-3d05ff00852c)
![image](https://github.com/user-attachments/assets/ddc4a7f7-c20c-4544-a0e5-1759692aee7c)
![image](https://github.com/user-attachments/assets/748c95ce-ad77-4be5-82f1-ff595995f39b)
![image](https://github.com/user-attachments/assets/117e98fc-f3dd-4baa-81bf-b0d04c6973e8)

## Summary

This PR introduces full internationalization (i18n) support, with French (FR) added as a second-language example. The i18n implementation is fully integrated with the current UI and production-ready.

---

### 1. 🧱 Introducing the i18n Framework

![image](https://github.com/user-attachments/assets/99300c4b-7f0e-4571-bfda-ff1e3c210d51)

Following the design proposed earlier, this PR completes the green-box portion in the diagram — enabling robust i18n support for the UI.

**What’s included:**
- Refactored existing language logic to use the new centralized framework.
- Added `/ui/i18n/`, which contains all configuration and utility functions for i18n.
- Locales are stored in `/assets/locale/` using a standardized JSON format, which can be reused by the Golang backend.
- All i18n development guidelines and documentation are now available under `/docs/`. The main `README.md` has been simplified and now links to the relevant sections.

---

### 2. 🌐 What’s Been Localized

- **Language toggle** implemented, with French (FR) added as a working second language.
- **Fallback mechanism**: If a key is missing in the selected locale, it will gracefully fall back to English (EN), which is the source of truth.
- Localized the **landing page** (`index.html`) and **simulator/spec page** (`index_template.html`).
- All **tab titles** (Gear, Settings, Talents, etc.) have been localized.
- The **sidebar**, including the header, buttons, and stat labels, is fully localized.

This implementation serves as a working example to guide future localization work.

---

### 3. 🧩 Design Considerations

#### 🗂️ Locale JSON Structure

Localization files are modularized by UI section to improve clarity and maintainability:

```
landing: for index.html
sim: for index_template.html
sidebar: includes header, buttons, and stats
common: shared labels like stats, classes, specs, etc.
gear
setting
talents
rotation
results
import
export
```

**Benefit**: Clear structure for contributors to locate and manage translations efficiently.

---

#### 🔁 Enum Mapping & Safe Translations

Many internal enums like `Strength` or `Agility` are used as UI text, keys, and logic values. Direct translation could break core functionality.

**Solution**:
- `entity_mapping.ts`: Maps raw Go-enum values to stable English keys.
- `translation.ts`: Maps those keys to localized strings from the locale files.

This separation ensures translation logic is safe, extensible, and won’t interfere with the sim's logic or state management.

---

1337LutZ commented:
The NaN / 0 is a WoWhead issue, can replicate it on their site as well.

1337LutZ commented:
@dodoels can you update your branch from current master

dodoels commented:
@1337LutZ please review when you get chance, thanks!

kayla-glick commented:
@1337LutZ can you send the page that you tested with on wowhead? Trying to pass it along to their devs

dodoels commented:
Hey @1337LutZ please keep me posted on the review - I will continue the work in anothe branch

rosenrusinov commented:
@kayla-glick @1337LutZ @dodoels I believe this MR has very big value for us as there are a lot of people that could benefit from the wowhead items being in their language (I was surprised how many people play the game in their native languages).

Is it possible we come back to this and try to get it in a mergable state?

dodoels commented:
> @kayla-glick @1337LutZ @dodoels I believe this MR has very big value for us as there are a lot of people that could benefit from the wowhead items being in their language (I was surprised how many people play the game in their native languages).
> 
> Is it possible we come back to this and try to get it in a mergable state?

Updated the branch with the latest changes. This should now be a good MVP to be included in the release.

dodoels commented on ui/core/components/character_stats.tsx:
```diff
@@ -51,17 +53,24 @@ export class CharacterStats extends Component {
-			const statName = unitStat.getShortName(player.getClass());
+			let statName: string;
+			if (unitStat.isStat()) {
+				statName = translateStat(unitStat.getStat());
```
To be specific, `enums` to `string` is more robust than `string` to `string` which requires more manual works

dodoels replied:
Resolved, but not sure if this is still revelant

```
// Not sure if this is needed anymore, but keeping it for now
if (playerClass == Class.ClassHunter) {
	return fullName.replace('Physical', 'Ranged');
} else {
	return fullName.replace('Physical', 'Melee');
}
```

I left it as it is for now

dodoels replied:
Resolved in https://github.com/wowsims/mop/pull/41/commits/e7f267cabfaf9b66b10ac57b14de6f42afff2cd4

dodoels commented:
@1337LutZ Addressed all your comments. Please checkout the branch and play with it see if any further issue. I will be afk until August, if this is not mergeble I will have to come back later - Otherwise, please feel free to merge it!
